Normalize spaces in union types in comments ("A | B" -> "A|B")

TysonAndre opened the pull request:
Mentioned in discussion of
https://github.com/Microsoft/tolerant-php-parser/pull/136/files 
(Couldn't find the code style guidelines referred to, if any. Is that just referring to the current state?)

Many IDEs and tools don't expect whitespace, and don't handle this.
So, they may act as if only the first type.

- https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md#appendix-a-types
  is a proposal, and is vague about how whitespace is handled
  outside `<>`
- https://github.com/phpDocumentor/phpDocumentor2/blob/develop/docs/references/phpdoc/tags/var.rst
  is vague
- https://github.com/phpDocumentor/ReflectionDocBlock/blob/master/src/DocBlock/Tags/Var_.php
  doesn't expect any whitespace after `@var`

msftclas commented:
@__TysonAndre__,
Thanks for your contribution.
To ensure that the project team has proper rights to use your work, please complete the Contribution License Agreement at https://cla.microsoft.com.

It will cover your contributions to all Microsoft-managed open source projects.
_Thanks,
Microsoft Pull Request Bot_

roblourens commented:
Thanks! I don't think the test files supposed to be changed?

TysonAndre commented:
Reverted the test files. I'm not sure when those changed, must have modified them somehow or checked out a stale version by accident.

After this was approved, `@param` and `@return` were also modified, and DiagnosticsProvider.php had phpdoc added and reordered.

msftclas commented:
__@TysonAndre__, thanks for signing the contribution license agreement. We will now validate the agreement and then the pull request.
<br />_Thanks, Microsoft Pull Request Bot_
